libathemecore/users.c

```diff
@@ -673,6 +675,13 @@ find_user_banned_channel(struct user *const restrict u, const char ban_type)
 	return NULL;
 }
 
+const char *
+format_hostmask(const struct user *user, const enum host_type type)
+{
+	sprintf(hostmaskbuf, "%s!%s!%s", user->nick, user->user, user->vhost);
```

Comment:
this currently just does `user->vhost` without checking the requested host type

i'd probably use `snprintf` with `sizeof hostmaskbuf` here for safety